pkg/evaluator/evaluator.go

```diff
@@ -639,6 +639,9 @@ func evalBinaryArrayExpr(op parser.Operator, left, right *Array) (Value, error)
 		return nil, fmt.Errorf("%w (array): %v", ErrOperation, op.String())
 	}
 	result := left.Copy()
+	if result.T == parser.GENERIC_ARRAY {
```

Comment:
Instead of doing this here for this one specific case, I wonder if it might be better to do this in `evalBinaryExpr` and set the type of the `Value` to that of the `*parser.BinaryExpression`, since you set that correctly in the parser. That way you don't need special knowledge of arrays in the evaluator and constrain that logic just to the parser.

Reply:
yes, conceptually I agree, this would be nicer and less land-mine-y.
however, iiuc, that'd  mean that I have extend the `Value` interface by a `SetType(t *parser.Type)` method I think which might not be worth it (especially as code size is a concern) - wdyt?